### PR TITLE
Update Dockerfile to run tests without root permissions

### PR DESCRIPTION
**Pull request by ethan-kusters**
## Summary

Fixes test failures that were occurring when using the included Dockerfile to run Swift-DocC's tests for Linux. The test suite does not expect to be run on a root user account. This also better emulates the way tests will be run on Linux in the Swift.org CI system.

The changes to the Dockerfile are based on the Dockerfiles used in swift-ci here: https://github.com/apple/swift-docker/blob/main/swift-ci/master/ubuntu/20.04/Dockerfile.

## Dependencies

None.

## Testing

Follow the instructions in the CONTRIBUTING documentation [here](https://github.com/ethan-kusters/swift-docc/blob/update-dockerfile-for-linux-testing/CONTRIBUTING.md#using-docker-to-test-swift-docc-for-linux) and confirm that all tests pass.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] ~~Added tests~~ N/A
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -209,28 +209,22 @@ by running the test suite in a Docker environment that simulates Swift on Linux.
 
 1. Install [Docker Desktop for Mac](https://www.docker.com/products/docker-desktop).
 
-2. Get the newest Swift Docker image with:
+2. Build Swift-DocC (see [Building Swift-DocC](#building-swift-docc)).
 
-    ```bash
-    docker pull swift
```
Comment (ethan-kusters) [inline CONTRIBUTING.md:215]: Pulling `swift` is not a prerequisite to building the Dockerfile so I've removed this step.

**Review by franklinsch — APPROVED** [review]
Looks great, thanks!

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Pull request merged**